docs/available-filters/envelope-encryption.adoc

```diff
@@ -0,0 +1,185 @@
+:kms-api-javadoc: https://javadoc.io/doc/io.kroxylicious/kroxylicious-kms/latest
+:encryption-api-javadoc: https://javadoc.io/doc/io.kroxylicious/kroxylicious-encryption
+
+= Envelope Encryption
+
+== What is it?
+
+A filter that provides a transparent https://kroxylicious.io/use-cases/[encryption-at-rest solution] for Apache Kafka.
+
+It works as follows:
+
+1. The filter intercepts the requests sent from the producing Kafka Client containing records.  The records
+   contained with the request are encrypted.  This happens before the records reach the Kafka Broker.
+2. The filter intercepts the responses sent from the Kafka Broker to the consuming Kafka Client.  The records
+   contained within the response are decrypted.  This happens before the records reach Kafka Client.
+
+The entire process is transparent from the point of view of both Kafka Client and Kafka Broker.  Neither are
+aware that the records are being encrypted.  Neither client nor broker have any access to the encryption keys.
+
+The filter encrypts the kafka records using a symmetric encryption key. It is possible to use different encryption
+keys for different topics. The encryption keys are held within by a Key Management Service (KMS).  It is the user's
```

Comment:
"The encryption keys" this bit isn't correct. The KEKs are, yes, but the encrypted DEKs are stored in Kafka. I know you're aiming for a simple explanation here. But I don't think it's helpful to give people an incorrect explanation. We should take the time and space to explain it properly, using pictures if that helps.

Reply:
I've added summary describing how envelope encryption is applied to the problem.  I've tried to keep it high level and reference the design for the details.